ulwgl_run.py

```diff
@@ -327,8 +334,16 @@ def main() -> None:  # noqa: D103
         os.environ[key] = val
 
     build_command(env, command, opts)
-    subprocess.run(command)
+    proc = subprocess.run(command)
+    return proc.returncode
 
 
 if __name__ == "__main__":
-    main()
+    try:
+        ret = main()
+    except Exception as e:  # noqa: BLE001
+        traceback.print_tb(e.__traceback__, file=sys.stderr)
+        print(f"{e}", file=sys.stderr)
+        sys.exit(1)
+    else:
+        sys.exit(ret)
```

Comment:
Please correct me if I am wrong, but `print_tb` just formats the traceback nicely compared to simply raising the exception, correct?

Reply:
Yes and no, `print_tb` is the way to print the traceback for the exception object, but ideally here, if logging is implemented, the `print_tb` and `print` calls should be replaced with `logging.excpetion`